packages/wonder-blocks-dropdown/components/dropdown-launcher.js

```diff
@@ -0,0 +1,244 @@
+// @flow
+
+import * as React from "react";
+import ReactDOM from "react-dom";
+import {StyleSheet} from "aphrodite";
+import type {AriaProps, StyleType} from "@khanacademy/wonder-blocks-core";
+import Dropdown from "./dropdown.js";
+import type {Item, DropdownItem} from "../util/types.js";
+import ActionItem from "./action-item.js";
+import OptionItem from "./option-item.js";
+
+type Props = {|
+    ...AriaProps,
+
+    /*
+     * The callback function used to open the Dropdown.
+     */
+    children: (openDropdown: () => void) => React.Element<any>,
+
+    opened?: boolean,
+
+    /*
+     * Text for the opener of this menu.
+     */
+    menuText: string,
+
+    /*
+     * The items present in the dropdown.
+     */
+    menuItems: Array<Item> | Item,
+
+    /*
+     * A callback that returns items that are newly selected. Use only if this
+     * menu contains select items (and make sure selectedValues is defined).
+     */
+    onChange?: (selectedItems: Array<string>) => mixed,
+
+    /*
+     * The values of the items that are currently selected. Use only if this
+     * menu contains select items (and make sure onChange is defined).
+     */
+    selectedValues?: Array<string>,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * opener component. Defaults to left-aligned.
+     */
+    alignment: "left" | "right",
+
+    /**
+     * Whether this component is disabled. A disabled dropdown may not be opened
+     * and does not support interaction. Defaults to false.
+     */
+    disabled: boolean,
+
+    /*
+     * Optional styling to add to the opener component wrapper.
+     */
+    style?: StyleType,
+
+    /*
+     * Test ID used for e2e testing.
+     */
+    testId?: string,
+
+    /*
+     * Optional styling to add to the dropdown wrapper.
+     */
+    dropdownStyle?: StyleType,
+|};
+
+type State = {|
+    /**
+     * Whether or not the dropdown is currently open.
+     */
+    open: boolean,
+
+    /**
+     * Whether or not last open state change was triggered by a keyboard click.
+     */
+    keyboard?: boolean,
+|};
+
+export default class DropdownLauncher extends React.Component<Props, State> {
+    openerElement: ?HTMLElement;
+
+    static defaultProps = {
+        alignment: "left",
+        disabled: false,
+    };
+
+    constructor(props: Props) {
+        super(props);
+        this.state = {
+            open: false,
+            keyboard: false,
+        };
+    }
+
+    handleItemSelected = () => {
+        // Bring focus back to the opener element.
+        if (this.openerElement) {
+            this.openerElement.focus();
+        }
+
+        this.setState({
+            open: false, // close the menu upon selection
+        });
+    };
+
+    handleOpenChanged = (open: boolean, keyboard?: boolean) => {
+        this.setState({
+            open,
+            keyboard,
+        });
+    };
+
+    handleOptionSelected = (selectedValue: string) => {
```

Comment:
Can you ask Nefaur which menu items need to be available in custom dropdown (we might not need this)?